src/OrleansTestKit/Streams/TestStream.cs

```diff
@@ -97,7 +105,7 @@ public Task OnNextBatchAsync(IEnumerable<T> batch, StreamSequenceToken token = n
 
         public Task<IList<StreamSubscriptionHandle<T>>> GetAllSubscriptionHandles()
         {
-            throw new NotImplementedException();
+            return Task.FromResult<IList<StreamSubscriptionHandle<T>>>(_handlers);
```

Comment:
When multiple calls are made to `GetAllSubscriptionHandles`, both task results will point to the same list reference. This doesn't appear to match the Orleans behavior (searching for `GetAllSubscriptionHandles` and `GetAllSubscriptions`, they return new lists), and it could lead to concurrency exceptions in some test scenarios.

```suggestion
            return Task.FromResult<IList<StreamSubscriptionHandle<T>>>(new List<StreamSubscriptionHandle<T>>(_handlers));
```

Would you also add a second unit test that demonstrates this behavior? It would also serve as a good regression test.

Reply:
Done.